File: app/src/main/java/com/kaciula/archiman/data/DataRepositoryModule.java
package com.kaciula.archiman.data;

import com.kaciula.archiman.data.local.database.UsersLocalDataSource;
import com.kaciula.archiman.data.remote.StackExchangeApi;
import com.kaciula.archiman.data.remote.UsersRemoteDataSource;
import com.kaciula.archiman.domain.boundary.data.AppInfoRepository;
import com.kaciula.archiman.domain.boundary.data.UserRepository;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module
public class DataRepositoryModule {

  @Singleton
  @Provides
  UserRepository providerUsersRepository(StackExchangeApi stackExchangeApi) {
    return new UserRepositoryImpl(new UsersRemoteDataSource(stackExchangeApi),
        new UsersLocalDataSource());
  }

  @Provides
  @Singleton
  AppInfoRepository provideAppInfoRepository() {
    return new AppInfoRepositoryImpl();
  }
}

File: app/src/main/java/com/kaciula/archiman/data/DataRepositoryModule.kt
package com.kaciula.archiman.data

import com.kaciula.archiman.data.local.database.UserLocalDataStore
import com.kaciula.archiman.data.remote.StackExchangeApi
import com.kaciula.archiman.data.remote.UserRemoteDataStore
import com.kaciula.archiman.domain.boundary.data.AppInfoRepository
import com.kaciula.archiman.domain.boundary.data.UserRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DataRepositoryModule {

    @Singleton
    @Provides
    fun providerUsersRepository(stackExchangeApi: StackExchangeApi): UserRepository {
        return UserRepositoryImpl(UserRemoteDataStore(stackExchangeApi),
                UserLocalDataStore())
    }

    @Provides
    @Singleton
    fun provideAppInfoRepository(): AppInfoRepository {
        return AppInfoRepositoryImpl()
    }
}
